### DISCUSSION TITLE
Vue version of `react-markdown`?

### DISCUSSION THREAD

**Discussion by SeryiBaran** [remarkjs/remark#1154] [Q&A] [answered]
Will there be something similar to react-markdown for Vue.js?

**Comment by wooorm** [comment, marked as answer]
No, I am not currently interested in doing that. Although, there probably already is a lot out there, that does this.

If I were to make this, I would use the same internals that this project uses (remark, rehype, etc), and then use `toJsxRuntime` from `hast-util-to-jsx-runtime` to generate Vue nodes: https://github.com/syntax-tree/hast-util-to-jsx-runtime/tree/main#example-vue

**Reply by Cat1007** [reply]
I have developed a package using this idea that I hope will help

https://www.npmjs.com/package/vue-markdown-next

**Comment by doroved** [comment]
> No, I am not currently interested in doing that. Although, there probably already is a lot out there, that does this.
> 
> If I were to make this, I would use the same internals that this project uses (remark, rehype, etc), and then use `toJsxRuntime` from `hast-util-to-jsx-runtime` to generate Vue nodes: https://github.com/syntax-tree/hast-util-to-jsx-runtime/tree/main#example-vue

Hi, it's a pity that there is no such library for vue. I need to be able to render markdown from a vue 3 component. Here is an example of how I would like to see it. Do you have any thoughts on how this could be implemented using marked?
https://github.com/markedjs/marked/issues/2915

**Reply by ChristianMurphy** [reply]
@SeryiBaran please actually read the link you included in your quote

> it's a pity that there is no such library for vue.

There is, @wooorm linked to it, and you included it in your question: https://github.com/syntax-tree/hast-util-to-jsx-runtime/tree/main#example-vue
You are welcome to create a higher level of abstraction if you want, but the key piece, a library that can render to Vue already exists.

> Do you have any thoughts on how this could be implemented using marked?

No, this isn't the `marked` community.
There are ways to implement this is `remark` and those are linked in the already selected answer.

**Comment by doroved** [comment]
> There are ways to implement this is remark and those are linked in the already selected answer.

Could you show a sample code using **remark** to render link to markdown using vue3 component?
Since it didn't work for me.

```js
import { renderToString } from '@vue/server-renderer'
import { Fragment, jsx, jsxs } from 'vue/jsx-runtime'
import { h } from 'hastscript'
import { toJsxRuntime } from 'hast-util-to-jsx-runtime'

  console.log(
    await renderToString(
      toJsxRuntime(h('CustomLink', { href: 'https://google.com', text: 'google' }), {
        Fragment,
        jsx,
        jsxs,
        elementAttributeNameCase: 'html'
      })
    )
  )
```

**Console.log**
```
<customlink href="https://google.com" text="google"></customlink>
```

**Reply by ChristianMurphy** [reply]
I'd suggest you start by reading about `unified`, `remark`, and `rehype` as this will involve all three
https://unifiedjs.com/learn/

The flow would be `remark-parse`, `remark-rehype`, then `rehype-vue` (a thin wrapper of `hast-util-to-jsx-runtime`)

> Since it didn't work for me.

Read the documentation for `hast-util-to-jsx-runtime`
Custom Vue components will only be rendered if you register them with the `components` option.
https://github.com/syntax-tree/hast-util-to-jsx-runtime/tree/main#options

Same goes for `rehype-vue`.

**Reply by doroved** [reply]
omg, It looks very cumbersome :(

**Reply by ChristianMurphy** [reply]
With a basic understanding of Vue and Unified, it is fairly streamlined, concise, and elegant.

If you have specific questions on how unified and the ecosystem work, feel free to ask them.

If you're here to complain other people aren't coding your project for you.
That isn't how open source works.
Contribute yourself, fund an effort, or buzz off.

**Reply by doroved** [reply]
No one asked you to write code, watch what you say.

**Reply by ChristianMurphy** [reply]
@doroved you have repeatably, chill.
Continue to be rude and you will be blocked/banned.

**Reply by doroved** [reply]
@ChristianMurphy How am I being rude? It seems like you're being rude to me now!
Are you offended that your solution, where you have to install 4 libraries, looks very cumbersome? That's funny.

**Reply by ChristianMurphy** [reply]
@doroved going back the the [first response in the thread](https://github.com/orgs/remarkjs/discussions/1154#discussioncomment-5753412). There isn't a single line abstraction, though anyone is welcome to create one.
After reading that you start badgering the maintainers to do so for you, [first in a completely unrelated library](https://github.com/orgs/remarkjs/discussions/1154#discussioncomment-6622973), [then with this library](https://github.com/orgs/remarkjs/discussions/1154#discussioncomment-6623266), [then complaining that understanding Vue and writing a few lines of code on your own is too cumbersome](https://github.com/orgs/remarkjs/discussions/1154#discussioncomment-6623706).

I appreciate that you are new to unified and perhaps to open source as a whole, and that it can be a lot to take in and process.
The way your are approaching it currently is trolling behavior and is unwelcome in the community.

If you would like support, follow the [support guidelines](https://github.com/remarkjs/.github/blob/main/support.md) and be respectful of the volunteers helping you.
Otherwise, move along.

**Reply by doroved** [reply]
What the hell did you write? Are you okay? Do you need any help?

**Comment by SeryiBaran** [comment]
@doroved , @ChristianMurphy , please, stop flood =(
